src/lib/homestore/heap_chunk_selector.cpp

```diff
@@ -0,0 +1,116 @@
+#include "heap_chunk_selector.h"
+
+#include <execution>
+#include <algorithm>
+#include <sisl/logging/logging.h>
+
+namespace homeobject {
+
+//https://github.com/facebook/folly/blob/61c11d77eb9a8bdc60f673017fccfbe900125cb6/folly/AtomicHashMap.h#L42
+//AtomichashMap has a max size limit of ~18x initial size.
+
+HeapChunkSelector::HeapChunkSelector() : 
+m_pdev_heap_map(PdevHeapMap(pdev_atomicmap_init_size)), 
+m_pdev_avalable_blk_map(pdev_atomicmap_init_size), 
+m_chunks(chunk_atomicmap_init_size) {}
+
+HeapChunkSelector::HeapChunkSelector(const uint32_t& pdev_heap_map_initial_size, 
+const uint32_t& pdev_avalable_blk_map_initial_size, const uint32_t& chunk_initial_num) : 
+m_pdev_heap_map(PdevHeapMap(pdev_heap_map_initial_size)), 
+m_pdev_avalable_blk_map(PdevAvalableBlkMap(pdev_avalable_blk_map_initial_size)),
+m_chunks(ChunkMap(chunk_initial_num)) {}
+
+
+//add_chunk may be called in Homeobject(when sealing a shard) or in Homestore(when adding chunks to vdev) concurently
+void HeapChunkSelector::add_chunk(csharedChunk& chunk) {
+    VChunk vchunk(chunk);
+    auto&& pdevID = vchunk.get_pdev_id();
+    auto&& [it, _] = m_pdev_heap_map.emplace(pdevID, std::make_shared<std::pair<std::mutex, VChunkHeap>>());
```

Comment:
What style of casing are we using, camelCase or underscore_case. It seems like code has both these cases in same method, would prefer consistency one way or other. Let us pick one as a team and follow.

Reply:
I think we have yet to discuss this. :)

My preference (let it be known) is for Classes and Structs to use CamelCase; but internal members to use snake_case. Parameters to functions should also prefer snake_case.